parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputFormat.java

```diff
@@ -145,6 +147,18 @@
   public static final String MAX_ROW_COUNT_FOR_PAGE_SIZE_CHECK = "parquet.page.size.row.check.max";
   public static final String ESTIMATE_PAGE_SIZE_CHECK = "parquet.page.size.check.estimate";
 
+  /**
+   * Used to override the value writer factory used to create Values Writers.
+   * This allows users to plug in their own factory class to create custom ValuesWriters rather
+   * use the Parquet provided default - {@link org.apache.parquet.column.values.factory.DefaultValuesWriterFactory}.
+   * Factory classes provided must implement {@link org.apache.parquet.column.values.factory.ValuesWriterFactory}.
+   *
+   * For e.g. "parquet.writer.factory-override" = "org.apache.parquet.column.values.factory.MyValuesWriterFactory"
+   * will ensure that we use MyValuesWriterFactory to create ValuesWriters for writing out data.
+   * See the documentation for ValuesWriterFactory for more details.
+   */
+  public static final String WRITER_FACTORY_OVERRIDE = "parquet.writer.factory-override";
```

Comment:
how about `parquet.write.value.writer.factory` -- no need to clarify that it's an override